harvesters/src/main/java/org/fao/geonet/kernel/harvest/harvester/geonet/Aligner.java

```diff
@@ -646,78 +648,82 @@ private void updateMetadata(final RecordInfo ri, final String id, final boolean
-                if (version != null && version.equals(MEFLib.Version.V2)) {
-                    fileType = "mef2";
-                }
-
-                IVisitor visitor = fileType.equals("mef2") ? new MEF2Visitor() : new MEFVisitor();
-
-                MEFLib.visit(mefFile, visitor, new IMEFVisitor()
-				{
-					public void handleMetadata(Element mdata, int index) throws Exception
-					{
-						md[index] = mdata;
-					}
-
-					//-----------------------------------------------------------------
-					
-					public void handleMetadataFiles(DirectoryStream<Path> files, Element info, int index) throws Exception
-					{
-                        // Import valid metadata
-                        Element metadataValidForImport = extractValidMetadataForImport(files, info);
-
-                        if (metadataValidForImport != null) {
-                            handleMetadata(metadataValidForImport, index);
-					}
+            if(!useChangeDate || ri.isMoreRecentThan(localChangeDate)) {
+    			Path mefFile = retrieveMEF(ri.uuid);
+    
+    			try
+    			{
+                    String fileType = "mef";
+                    MEFLib.Version version = MEFLib.getMEFVersion(mefFile);
+                    if (version != null && version.equals(MEFLib.Version.V2)) {
+                        fileType = "mef2";
                     }
-					
-					public void handleInfo(Element info, int index) throws Exception
-					{
-						updateMetadata(ri, id, md[index], info, localRating);
-						publicFiles[index] = info.getChild("public");
-						privateFiles[index] = info.getChild("private");
-					}
-
-					//-----------------------------------------------------------------
-
-					public void handlePublicFile(String file, String changeDate, InputStream is, int index) throws IOException
-					{
-						updateFile(id, file, "public", changeDate, is, publicFiles[index]);
-					}
-
-					public void handleFeatureCat(Element md, int index)
-							throws Exception {
-						// Feature Catalog not managed for harvesting
-					}
-
-					public void handlePrivateFile(String file,
-							String changeDate, InputStream is, int index)
-							throws IOException {
-	                       updateFile(id, file, "private", changeDate, is, privateFiles[index]);
-					}
-					
-				});
-			}
-			catch(Exception e)
-			{
-				//--- we ignore the exception here. Maybe the metadata has been removed just now
-				result.unretrievable++;
-			}
-			finally
-			{
-                try{
-                    Files.deleteIfExists(mefFile);
-                } catch (IOException e) {
-	                 log.warning("Unable to delete mefFile: "+mefFile);
-	             }
-
-			}
+    
+                    IVisitor visitor = fileType.equals("mef2") ? new MEF2Visitor() : new MEFVisitor();
+    
+                    MEFLib.visit(mefFile, visitor, new IMEFVisitor()
+    				{
+    					public void handleMetadata(Element mdata, int index) throws Exception
```

Comment:
Not critical but I would be amiss not to mention it :-)   Tabs should be spaces if I remember the decision correctly